src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,369 @@
+      //   2. we have no sensors sensed as we are in between the intake and the drum
+      if (inputs.isDrumIRBlocked) {
+        intakeState = IntakeState.NOTE_IN_INTAKE_AND_DRUM;
+        this.transitionGamePiece();
+      } else if (inputs.isLeftRollerIRBlocked || inputs.isRightRollerIRBlocked) {
+        intakeState = IntakeState.NOTE_IN_INTAKE;
+      } else {
+        // we are now in between intake and drum
+        intakeState = IntakeState.IN_BETWEEN_INTAKE_AND_DRUM;
+
+        // set the timeout here
+        intakeAndDrumTimeout = 0;
+      }
+    } else if (intakeState == IntakeState.IN_BETWEEN_INTAKE_AND_DRUM) {
+      //  the only possible next step is that we are only in the drum
+      //  add a timeout so that if we are here for too long we go back to being empty
+      intakeAndDrumTimeout++;
+
+      if (inputs.isDrumIRBlocked) {
+        intakeState = IntakeState.NOTE_IN_DRUM;
+        this.transitionGamePiece();
+      } else if (intakeAndDrumTimeout >= IntakeConstants.IN_BETWEEN_TIMEOUT_SECONDS * 50) {
+        intakeState = IntakeState.EMPTY;
+        leds.setIntakeLEDState(IntakeLEDState.WAITING_FOR_GAME_PIECE);
+      }
+    } else if (intakeState == IntakeState.NOTE_IN_INTAKE_AND_DRUM) {
+      // the only possible next step is that we are only in the drum
+      //    1. this is when the intake IR stops being detected
+      // assuming that we do not need to call transitionGamePiece() again because we already called
+      if (!(inputs.isLeftRollerIRBlocked || inputs.isRightRollerIRBlocked)) {
+        intakeState = IntakeState.NOTE_IN_DRUM;
+        this.repelGamePiece();
+      }
+    } else if (intakeState == IntakeState.NOTE_IN_DRUM) {
+      // the only possible next step is becoming empty between the drum and the kicker
+      if (!inputs.isDrumIRBlocked) {
+        intakeState = IntakeState.IN_BETWEEN_DRUM_AND_KICKER;
+        this.turnTransitionOff();
+        this.repelGamePiece();
+      }
+    } else if (intakeState == IntakeState.NOTE_IN_KICKER) {
+      // the only possible next step is that we become empty after shooting
+      if (!inputs.isKickerIRBlocked) {
+        intakeState = IntakeState.EMPTY;
+        leds.setIntakeLEDState(IntakeLEDState.WAITING_FOR_GAME_PIECE);
+        this.intakeGamePiece();
+      }
+    }
+  }
+
+  public Command getSystemCheckCommand() {
+    /*
+     * Order of what will be tested
+     * 1. make sure everything works at right velocities
+     * 2. then check that all the states it goes through when it gets a note are correct  
+     *
+     */
+    return Commands.sequence(
+      getVelocitiesCheckCommand(),
+      getStatesCheckCommand()
+    ).until(() -> !FaultReporter.getInstance().getFaults(SUBSYSTEM_NAME).isEmpty())
+    .andThen(Commands.runOnce(this::turnIntakeOff))
+    .withName(SUBSYSTEM_NAME + "SystemCheck");
+  }
+
+
+  private Command getVelocitiesCheckCommand() {
+    return Commands.parallel(
+      Commands.run(this::runIntakeStateMachine),
```

Comment:
This method is already called in periodic; so, you don't need to call it again. I think you need to explicitly set the state to EMPTY and invoke intakeGamePiece and transitionGamePiece, however.